huntsman/camera/pyro.py

```diff
@@ -489,8 +487,62 @@ def _async_wait(self, future_result, name='?', event=None, timeout=None):
             raise error.Timeout(msg)
 
         return result
+    
 
-
+    def _process_fits(self, file_path, info, *args, **kwargs):
+        '''
+        Override _process_fits, called by process_exposure in take_observation.
+        
+        The difference is that we do an NGAS push following the processing.
+        '''
+        #Call the super method
+        result = super()._process_fits(file_path, info, *args, **kwargs)
+        
+        #Do the NGAS push
+        self._NGASpush(file_path, info)
+        
+        return result
+        
+    
+    def _NGASpush(self, filename, metadata, filename_ngas=None, port=7778):
+        '''
+        Parameters
+        ----------
+        filename (str):
+            The local filename, the basename of which is the default NGAS
+            filename.
+        filename_ngas (str, optional):
+            The NGAS filename. If None, auto-assign based on filename.
+        port (int, optional):
+            The port of the NGAS server. Defaults to the TCP port.
+            
+        '''        
+        #Define the NGAS filename 
+        if filename_ngas is None:
+            extension = os.path.splitext(filename)[1] 
+            filename = f"{metadata['image_id']}{extension}"
```

Comment:
Huh. Yeah, that'll do if all we want is for the filename to be unique and somewhat identifiable. Might run into trouble down the track if we get high frame rate stuff going because of the low resolution of the timestamp in `image_id`, though. That will also be a problem for the on disc filenames too.

Reply:
Yes, I think that could be addressed by modifying the definition of `image_id` in the future.